tests/python/shared/fixtures/init.py

```diff
@@ -240,11 +240,26 @@ def kube_restore_clickhouse_db():
 
 
 def docker_restore_redis_inmem():
-    docker_exec_redis_inmem(["redis-cli", "-e", "flushall"])
+    docker_exec_redis_inmem(
+        [
+            "sh",
+            "-c",
+            "for p in rq:finished:* rq:job:* rq:wip:* rq:finished:* rq:failed:*; "
+            'do redis-cli -e --scan --pattern "$p" | xargs -r redis-cli -e del ; done',
```

Comment:
Could you explain what you did here?

Reply:
Previously we reset the entire Redis DBs including RQ related keys about registered workers, which would cause the worker to restart on the next heartbeat, but in fact to pass the tests we only need to clear the keys related with jobs and statuses.
Here I find all keys related to jobs and their status and then delete them without deleting the whole DB

Reply:
In that case, I don't think you got all the keys that need to be removed. E.g. there's the [scheduled jobs key](https://github.com/rq/rq-scheduler/blob/656898e31c9da2747639173fd5b672506b957f29/rq_scheduler/scheduler.py#L27), the [export cache lock keys](https://github.com/cvat-ai/cvat/blob/5f279771c7c245ec3fefea1778662a10580fbc33/cvat/apps/dataset_manager/util.py#L104), the [blocker keys](https://github.com/cvat-ai/cvat/blob/5f279771c7c245ec3fefea1778662a10580fbc33/cvat/utils/background_jobs.py#L33). There could be others I missed.

In general, I don't think this approach is workable, because it's very easy to forget to update this function when writing code that uses new Redis keys. I think you should do the opposite; keep certain keys and delete everything else. And please add a comment to explain what the code is doing.